agent/connect/x509_patch.go

```diff
@@ -0,0 +1,122 @@
+package connect
+
+import (
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"fmt"
+	"net"
+	"net/url"
+	"unicode"
+)
+
+// NOTE: the contents of this file were lifted from
+// $GOROOT/src/crypto/x509/x509.go from a Go 1.16.5 checkout.
+//
+//
+// After https://go-review.googlesource.com/c/go/+/329129 lands in a Go release
+// we are compiling against we can safely remove all of this code.
+
+var (
+	x509_oidExtensionSubjectAltName = []int{2, 5, 29, 17}
+)
+
+const (
+	x509_nameTypeEmail = 1
+	x509_nameTypeDNS   = 2
+	x509_nameTypeURI   = 6
+	x509_nameTypeIP    = 7
+)
+
+func HackSANExtensionForCSR(template *x509.CertificateRequest) {
+	switch {
+	case len(template.DNSNames) > 0:
+	case len(template.EmailAddresses) > 0:
+	case len(template.IPAddresses) > 0:
+	case len(template.URIs) > 0:
+	default:
+		return
+	}
+
+	if x509_oidInExtensions(x509_oidExtensionSubjectAltName, template.ExtraExtensions) {
+		return
+	}
+
+	// Ensure we wipe the subject field completely since we're going to use the
+	// SAN field instead.
+	template.RawSubject = nil
+	template.Subject = pkix.Name{}
```

Comment:
This is my own amendment to the stdlib fix. I feel like at this point it's safe to remove this, but also if we aren't using the Subject fields why include them? Thoughts?

When I switch to the patched go toolchain and disable this entire hack everything works, so maybe i really don't need this.

Reply:
Removed